dist/resources/IMessageBlock.json

```diff
@@ -0,0 +1,116 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$ref": "#/definitions/IMessageBlock",
+  "definitions": {
+    "IMessageBlock": {
+      "type": "object",
+      "properties": {
+        "uuid": {
+          "type": "string",
+          "description": "A globally unique identifier for this Block.  (See UUID Format: https://floip.gitbook.io/flow-specification/flows#uuid-format)"
+        },
+        "name": {
+          "type": "string",
+          "description": "A human-readable \"variable name\" for this block. This must be restricted to word characters so that it can be used as a variable name in expressions. When blocks write results output, they write to a variable corresponding the name of the block."
+        },
+        "label": {
+          "type": "string",
+          "description": "A human-readable free-form description for this Block."
+        },
+        "semantic_label": {
+          "type": "string",
+          "description": "A user-controlled field that can be used to code the meaning of the data collected by this block in a standard taxonomy or coding system, * e.g.: a FHIR ValueSet, an industry-specific coding system like SNOMED CT, or an organization's internal taxonomy service. (e.g. \"SNOMEDCT::Gender finding\")"
+        },
+        "vendor_metadata": {
+          "type": "object",
+          "description": "A set of key-value elements that is not controlled by the Specification, but could be relevant to a specific vendor/platform/implementation."
+        },
+        "type": {
+          "type": "string",
+          "description": "A specific string designating the type or \"subclass\" of this Block. This must be one of the Block type names within the specification, such as Core.RunFlow or MobilePrimitives.Message."
+        },
+        "config": {
+          "$ref": "#/definitions/IMessageBlockConfig",
+          "description": "Additional parameters that are specific to the type of the block. Details are provided within the Block documentation."
+        },
+        "exits": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IBlockExit"
+          },
+          "description": "a list of all the exits for the block. Exits must contain the required keys below, and can contain additional keys based on the Block type"
+        }
+      },
+      "required": [
+        "config",
+        "exits",
+        "name",
```

Comment:
@bzabos , Json schema recognises empty string as valid (even we put it under `required`). Any idea how we can change our flow-runner interface to automatically generate constraints? eg:
- `non-empty` string for `name`s`
- positive number for `IFlow.interaction_timeout`

Reply:
this is resolved, I implemented json schema annotations